app/presenters/aggregated_tag_mapping_presenter.rb

```diff
@@ -0,0 +1,20 @@
+class AggregatedTagMappingPresenter < SimpleDelegator
+  def errored?
+    presented_tag_mappings.any? { |tag_mapping| tag_mapping.state == 'errored' }
+  end
+
+  def error_messages
+    presented_tag_mappings.flat_map do |tag_mapping|
+      next unless tag_mapping.message
+
+      # splits message on ".", but keeps it in the message.
+      tag_mapping.message.split(/(?<=[.])/)
+    end.reject(&:blank?)
+  end
+
+  def presented_tag_mappings
+    tag_mappings.map do |tag_mapping|
+      TagMappingPresenter.new(tag_mapping)
```

Comment:
Ok so the aggregated tag mapping object will have a number of TagMappings, so you still need this in order to show the label. That's fine then, nothing to do here 👍